tests/micro/arduino/test_arduino_workflow.py

```diff
@@ -49,6 +51,30 @@ def project_dir(workspace_dir):
     return workspace_dir / "project"
 
 
+# Saves the Arduino project's state, runs the test, then resets it
+@pytest.fixture(scope="function")
+def does_not_affect_state(project_dir):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        prev_project_state = pathlib.Path(temp_dir)
+        shutil.copytree(project_dir, prev_project_state / "project")
+        yield
+
+        # We can't delete project_dir or it'll mess up the Arduino CLI working directory
```

Comment:
could you explain why or what specific behavior deleting the directory triggers?